src/hotspot/share/opto/memnode.cpp

```diff
@@ -3233,6 +3267,15 @@ Node* MergePrimitiveStores::make_merged_input_value(const Node_List& merge_list)
          (_phase->type(merged_input_value)->isa_long() != nullptr && new_memory_size == 8),
          "merged_input_value is either int or long, and new_memory_size is small enough");
 
+  if (_value_order == DataOrder::Reverse) {
+    if (new_memory_size == 8) {
+      merged_input_value = _phase->transform(new ReverseBytesLNode(nullptr, merged_input_value));
+    } else if (new_memory_size == 4) {
+      merged_input_value = _phase->transform(new ReverseBytesINode(nullptr, merged_input_value));
+    } else {
+      return nullptr;
```

Comment:
```suggestion
      // <say why we cannot do anything for 2-element byte case>.
      return nullptr;
```

Reply:
Can you please put a comment here about the limitations?

Reply:
I will support 2-bytes reverse too. I'm not clear if I need ReverseBytesUS or ReverseBytesS here.